core/dbt/deps/tarball.py

```diff
@@ -0,0 +1,74 @@
+from dbt.contracts.project import RegistryPackageMetadata, TarballPackage
+from dbt.deps.base import PinnedPackage, UnpinnedPackage
+
+
+class TarballPackageMixin:
+    def __init__(self, tarball: str) -> None:
+        super().__init__()
+        self.tarball = tarball
+
+    @property
+    def name(self):
+        return self.tarball
+
+    def source_type(self) -> str:
+        return "tarball"
+
+
+class TarballPinnedPackage(TarballPackageMixin, PinnedPackage):
+    def __init__(self, tarball: str, package: str) -> None:
+        super().__init__(tarball)
+        # setup to recycle RegistryPinnedPackage fns
+        self.package = package
+        self.version = "tarball"
+
+    @property
+    def name(self):
+        return self.package
+
+    def get_version(self):
+        return self.version
+
+    def nice_version_name(self):
+        return f"Tarball (url: {self.tarball})"
```

Comment:
Can we have this be "tarball" instead of "Tarball"?

Reply:
changed, thanks! 5410f1f6188bd87efebadc86c7d8cd9545aca744